src/toil/utils/toilDebugFile.py

```diff
@@ -109,18 +117,35 @@ def main() -> None:
 
     if options.fetch:
         # Copy only the listed files locally
-        logger.debug("Fetching local files: %s", options.fetch)
-        fetchJobStoreFiles(jobStore=jobStore, options=options)
+
+        if isinstance(jobStore, FileJobStore):
+            logger.debug("Fetching local files: %s", options.fetch)
+            fetchJobStoreFiles(jobStore=jobStore, options=options)
+        else:
+            logger.critical("Can only fetch by name or glob from file-based job stores")
+            sys.exit(1)
```

Comment:
I didn't want this to raise  because I mean for this to be a user-facing error message, and I don't want to fail with a stack trace in response to expected user input. The user is likely to ask us to try to do this, we didn't mistakenly call unimplemented code.